packages/automation-contracts/scheduler/contracts/SchedulerGelatoResolver.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: AGPLv3
+// solhint-disable not-rely-on-time
+pragma solidity ^0.8.0;
+
+import { IFlowScheduler } from "./interface/IFlowScheduler.sol";
+import {
+    ISuperToken
+} from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperToken.sol";
+
+contract SchedulerGelatoResolver {
+
+    IFlowScheduler public canonicalScheduler;
+
+    constructor(address _canonicalScheduler) {
+        canonicalScheduler = IFlowScheduler(_canonicalScheduler);
+    }
+
+    /**
+     * @dev Gelato resolver that checks whether Flow Scheduler action can be taken
+     * @return bool whether there is a valid Flow Scheduler action to be taken or not
+     * @return bytes the function payload to be executed (empty if none)
+     */
+     function checker(
+        address superToken,
+        address sender,
+        address receiver
+    ) external view returns( bool , bytes memory ) {
+
+        IFlowScheduler.FlowSchedule memory flowSchedule = canonicalScheduler.getFlowSchedule(
+            superToken, 
+            sender, 
+            receiver
+        );
+
+        if (flowSchedule.endDate != 0 && block.timestamp >= flowSchedule.endDate) {
+
+            // return canExec as true and executeDeleteFlow payload
+            return (
+                true,
+                abi.encodeCall( IFlowScheduler.executeDeleteFlow,
+                    (
+                        ISuperToken(superToken),
+                        sender,
+                        receiver,
+                        "" // not supporting user data until encoding challenges are solved
+                    )
+                )
+            );
+
+        }
+
+        else if ( flowSchedule.startDate != 0 &&
```

Comment:
I feel like the condition for "create flow" should be before "delete flow".

Reply:
I was under the impression there could be a situation where flowSchedule.startDate + startMaxDelay is greater than flowSchedule.endDate and in that case, the resolver could return a start payload when it shoud be a delete payload if the condition for start came before delete

However, looking here: https://github.com/superfluid-finance/protocol-monorepo/blob/gelato-scheduler-automation/packages/automation-contracts/scheduler/contracts/FlowScheduler.sol#L158

I see that's not the case as would be blocked by flowSchedule.startDate != 0 on line 52. I can make this adjustment!